File: editorial/ruby/prob3-1-2.rb
# 入力
n = gets.to_i

# 素因数分解
answer = []
limit = Math.sqrt(n).to_i
(2..limit).each do |i|
  while n % i == 0
    n /= i
    answer << i
  end
end

answer << n if n >= 2

# 出力
puts answer.join(' ')

File: editorial/go/prob3-1-2.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var n int
	fmt.Scan(&n)

	answer := make([]int, 0)
	limit := int(math.Sqrt(float64(n)))
	for i := 2; i <= limit; i++ {
		for n%i == 0 {
			n /= i
			answer = append(answer, i)
		}
	}

	if n >= 2 {
		answer = append(answer, n)
	}

	for i, v := range answer {
		if i > 0 {
			fmt.Print(" ")
		}
		fmt.Print(v)
	}
	fmt.Println()
}
